src/node/history.h

```diff
@@ -576,22 +576,36 @@ namespace ccf
       replicated_state_tree.append(rh);
     }
 
-    bool verify_and_sign(
+    kv::TxHistory::Result verify_and_sign(
       PrimarySignature& sig, kv::Term* term = nullptr) override
     {
-      if (!verify(term))
+      if (!verify(term, &sig))
       {
-        return false;
+        return kv::TxHistory::Result::FAIL;
+      }
+
+      kv::TxHistory::Result result = kv::TxHistory::Result::OK;
+
+      auto progress_tracker = store.get_progress_tracker();
+      if (progress_tracker)
```

Comment:
Should `verify_and_sign()`return an error if `progress_tracker` is not set?

Reply:
Yes good catch. This should just not work if it progress_tracker is not set. This is an invalid setup.